### PR TITLE
[freetype] Update to 2.10.2

### PR DESCRIPTION
**Pull request by kevinlul**
https://sourceforge.net/projects/freetype/files/freetype2/2.10.2/

**Which triplets are supported/not supported? Have you updated the CI baseline?**

Need to verify that removing patch 5 actually works for dynamic triplets like `x86-windows` and `x64-windows`. In the FreeType CMake, we have
```cmake
if (WIN32)
  target_compile_definitions(
    freetype PRIVATE _CRT_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_WARNINGS)
  if (BUILD_SHARED_LIBS)
    target_compile_definitions(
      freetype PRIVATE DLL_EXPORT)
  endif ()
endif ()
```
so there shouldn't be a need to patch in a custom flag for this.

**Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?** Yes

### CODE REVIEW

**Review by LilyWangL — APPROVED** [review]

File: ports/freetype/portfile.cmake [hunk 1]
```diff
@@ -25,10 +24,10 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         png         CMAKE_DISABLE_FIND_PACKAGE_PNG
 )
 
-if(NOT ${VCPKG_LIBRARY_LINKAGE} STREQUAL "dynamic")
-    set(ENABLE_DLL_EXPORT OFF)
+if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
```
Comment (BillyONeal) [inline ports/freetype/portfile.cmake:27]: I believe this is already the default behavior, `vcpkg_configure_cmake` already has:

```
    if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
        list(APPEND _csc_OPTIONS -DBUILD_SHARED_LIBS=ON)
    elseif(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        list(APPEND _csc_OPTIONS -DBUILD_SHARED_LIBS=OFF)
    else()
        message(FATAL_ERROR
            "Invalid setting for VCPKG_LIBRARY_LINKAGE: \"${VCPKG_LIBRARY_LINKAGE}\". "
            "It must be \"static\" or \"dynamic\"")
    endif()
```

Comment (dan-shaw) [reply ports/freetype/portfile.cmake:27]: I agree with Billy. I think we can remove this change.

**Comment by kevinlul** [comment]
Is there anything that I need to do? Maybe rebase onto current `master`? Though it looks like CI is failing on that too.

**Comment by LilyWangL** [comment]
Waiting for #12067 merged, this PR will fix CI build error.

**Review by ras0219-msft — APPROVED** [review]

**Pull request merged**